.github/workflows/release.yml

```diff
@@ -57,12 +60,37 @@ jobs:
         dapper -f Dockerfile --target dapper make test
 
     - name: Publish Image Runtime
-      run: |
-        GITHUB_ACTION_TAG=${{ github.ref_name }} dapper -f Dockerfile --target dapper make publish-image-runtime
-      env:
-        DOCKER_USERNAME: ${{ env.DOCKER_USERNAME }}
-        DOCKER_PASSWORD: ${{ env.DOCKER_PASSWORD }}
-    
+      uses: rancher/ecm-distro-tools/actions/publish-image@master
+      with:
+        image: "rke2-runtime"
+        tag: ${{ github.ref_name }}
+        make-target: publish-image-runtime
+
+        public-repo: rancher
+        public-username: ${{ env.DOCKER_USERNAME }}
```

Comment:
Should we use secrets for all the sensitive data? 
```(yaml)
          public-repo: rancher
          public-username: ${{ secrets.DOCKER_USERNAME }}
          public-password: ${{ secrets.DOCKER_PASSWORD }}
          prime-repo: rancher
          prime-registry: ${{ secrets.PRIME_REGISTRY }}
          prime-username: ${{ secrets.PRIME_REGISTRY_USERNAME }}
          prime-password: ${{ secrets.PRIME_REGISTRY_PASSWORD }}
```

Reply:
We're getting those values from vault, the `"Read secrets"` step will retrieve them from vault and set it as `envs`. The Github CI will mask those, so no secrets leak.